include/itkCuberilleImageToMeshFilter.hxx

```diff
@@ -535,6 +549,81 @@ CuberilleImageToMeshFilter<TInputImage,TOutputMesh,TInterpolator>
     }
 }
 
+template<typename TInputImage, typename TOutputMesh, typename TInterpolator>
+size_t
+CuberilleImageToMeshFilter<TInputImage,TOutputMesh,TInterpolator>
+::CalculateBitmaskIDForVertexIndex( const IndexType &vindex ) {
+  typename IndexType::OffsetType ones;
+  ones.Fill(1);
+  const IndexType localorigin = vindex - ones;
+  std::bitset<8> bitmask;
+  for (size_t i = 0; i < 8; ++i) {
+    std::bitset<3> bits(i);
+    typename IndexType::OffsetType offset;
+    offset[0] = bits[0];
+    offset[1] = bits[1];
+    offset[2] = bits[2];
+    const auto index = localorigin + offset;
+    const auto pixel = this->GetInput()->GetPixel( index );
+    bitmask[i] = (pixel >= this->m_IsoSurfaceValue);
+  }
+  return static_cast<size_t>(bitmask.to_ulong());
+}
+
+template<typename TInputImage, typename TOutputMesh, typename TInterpolator>
+void
+CuberilleImageToMeshFilter<TInputImage,TOutputMesh,TInterpolator>
+::CalculateLabelsArray()
+{
+
+  using TImage = itk::Image<unsigned char, 3>;
+  using TConnected = itk::ConnectedComponentImageFilter< TImage, TImage >;
+
+  for (size_t mask = 0; mask < pow(2, 8); ++mask) {
+
+    std::bitset<8> bitmask(mask);
+
+    const auto image = TImage::New();
+    TImage::SizeType size;
+    size.Fill( 2 );
+
+    TImage::IndexType origin;
+    origin.Fill( 0 );
+
+    TImage::RegionType region(origin, size);
+
+    image->SetRegions( region );
+    image->Allocate();
+    image->FillBuffer( 0 );
+
+    for (size_t index = 0; index < pow(2, 3); ++index) {
+      std::bitset<3> bitindex(index);
+      image->SetPixel( {{bitindex[0], bitindex[1], bitindex[2]}}, bitmask[index] );
+    }
+
+    const auto connected = TConnected::New();
+    connected->SetInput( image );
+    connected->FullyConnectedOff();
```

Comment:
The construction can probably be taken out of the loop, and there should be small performance benefit to it.